python/ray/worker.py

```diff
@@ -1683,6 +1706,20 @@ def _init(address_info=None,
     # shutdown() when the Python script exits.
     if driver_mode == LOCAL_MODE:
         driver_address_info = {}
+    elif driver_mode == EXTERNAL_CLIENT_MODE and use_raylet:
+        driver_address_info = {
+            "node_ip_address": address_info["node_ip_address"],
+            "redis_address": address_info["redis_address"],
+            "gateway_socat_port": gateway_socat_port,
+            "gateway_data_port": gateway_data_port,
+            "raylet_socket_name": address_info["raylet_socket_names"][0]
```

Comment:
This is failing for me when I try to connect to an external client to a remote machine, because `address_info` which is returned by `get_address_info_from_redis` consists of the address info for the processes on the local machine, but there are none on the local machine.

This does work for me if I attach the external client from the same node that `ray start` was run on.

Am I using it incorrectly or something?

Reply:
I thought the issue might be that I'm trying this on AWS, where nodes have distinct public and private IP addresses, but I don't think that's the issue.

Reply:
Yeah, something fishy is going on here. Will have a look. Tests are passing for me, so need to reproduce this.